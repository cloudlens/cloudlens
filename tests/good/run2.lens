lens l1 (log) {
     process (entry in log.session) {
         print("L1 " + entry.message)
     }
}

lens l2 (log) {
     process (entry in log.session) {
         print("L2 " + entry.message)
     }
}

lens l3 (log) {
     process (entry in log.session) {
         print("L3 " + entry.message)
     }
}

lens l4 (log) {
     process (entry in log.session) {
         print("L4 " + entry.message)
     }
}


match (
    "Session (?<date>\d\d-\d\d-\d\d\d\d)"   
)

group (entry) upon entry.date in session {
    "^"
}

process (entry in log.session) {
    print("1 " + entry.message)
}

process (entry in log.session) {
    print("2 " + entry.message)
}

l1()
l2()

process (entry in log.session) {
    print("3 " + entry.message)
}

l3()
l4()

