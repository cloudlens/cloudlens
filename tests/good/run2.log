Session 06-26-2016
Welcome back!
Process emacs, execution time 123s, status success
Process eclipse, execution time 4000s, status success
Process cloudlens, execution time 2s, status failure
Process firefox, execution time 6899s, status success
Bye!
Session 06-27-2016
Welcome back!
Process mail, execution time 3233s, status success
Process firefox, execution time 12049s, status sucess
Bye!
Session 06-28-2016
Welcome back!
Process cloudlens, execution time 66s, status success
Process firefox, execution time 8762s, status success
Process eclipse, execution time 2390s, status success
Process cloudlens, execution time 2s, status failure
Process cloudlens, execution time 5s, status sucess
Bye!