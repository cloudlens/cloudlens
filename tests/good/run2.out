1 Session 06-26-2016
2 Session 06-26-2016
L1 Session 06-26-2016
L2 Session 06-26-2016
3 Session 06-26-2016
L3 Session 06-26-2016
L4 Session 06-26-2016
1 Welcome back!
2 Welcome back!
L1 Welcome back!
L2 Welcome back!
3 Welcome back!
L3 Welcome back!
L4 Welcome back!
1 Process emacs, execution time 123s, status success
2 Process emacs, execution time 123s, status success
L1 Process emacs, execution time 123s, status success
L2 Process emacs, execution time 123s, status success
3 Process emacs, execution time 123s, status success
L3 Process emacs, execution time 123s, status success
L4 Process emacs, execution time 123s, status success
1 Process eclipse, execution time 4000s, status success
2 Process eclipse, execution time 4000s, status success
L1 Process eclipse, execution time 4000s, status success
L2 Process eclipse, execution time 4000s, status success
3 Process eclipse, execution time 4000s, status success
L3 Process eclipse, execution time 4000s, status success
L4 Process eclipse, execution time 4000s, status success
1 Process cloudlens, execution time 2s, status failure
2 Process cloudlens, execution time 2s, status failure
L1 Process cloudlens, execution time 2s, status failure
L2 Process cloudlens, execution time 2s, status failure
3 Process cloudlens, execution time 2s, status failure
L3 Process cloudlens, execution time 2s, status failure
L4 Process cloudlens, execution time 2s, status failure
1 Process firefox, execution time 6899s, status success
2 Process firefox, execution time 6899s, status success
L1 Process firefox, execution time 6899s, status success
L2 Process firefox, execution time 6899s, status success
3 Process firefox, execution time 6899s, status success
L3 Process firefox, execution time 6899s, status success
L4 Process firefox, execution time 6899s, status success
1 Bye!
2 Bye!
L1 Bye!
L2 Bye!
3 Bye!
L3 Bye!
L4 Bye!
1 Session 06-27-2016
2 Session 06-27-2016
L1 Session 06-27-2016
L2 Session 06-27-2016
3 Session 06-27-2016
L3 Session 06-27-2016
L4 Session 06-27-2016
1 Welcome back!
2 Welcome back!
L1 Welcome back!
L2 Welcome back!
3 Welcome back!
L3 Welcome back!
L4 Welcome back!
1 Process mail, execution time 3233s, status success
2 Process mail, execution time 3233s, status success
L1 Process mail, execution time 3233s, status success
L2 Process mail, execution time 3233s, status success
3 Process mail, execution time 3233s, status success
L3 Process mail, execution time 3233s, status success
L4 Process mail, execution time 3233s, status success
1 Process firefox, execution time 12049s, status sucess
2 Process firefox, execution time 12049s, status sucess
L1 Process firefox, execution time 12049s, status sucess
L2 Process firefox, execution time 12049s, status sucess
3 Process firefox, execution time 12049s, status sucess
L3 Process firefox, execution time 12049s, status sucess
L4 Process firefox, execution time 12049s, status sucess
1 Bye!
2 Bye!
L1 Bye!
L2 Bye!
3 Bye!
L3 Bye!
L4 Bye!
1 Session 06-28-2016
2 Session 06-28-2016
L1 Session 06-28-2016
L2 Session 06-28-2016
3 Session 06-28-2016
L3 Session 06-28-2016
L4 Session 06-28-2016
1 Welcome back!
2 Welcome back!
L1 Welcome back!
L2 Welcome back!
3 Welcome back!
L3 Welcome back!
L4 Welcome back!
1 Process cloudlens, execution time 66s, status success
2 Process cloudlens, execution time 66s, status success
L1 Process cloudlens, execution time 66s, status success
L2 Process cloudlens, execution time 66s, status success
3 Process cloudlens, execution time 66s, status success
L3 Process cloudlens, execution time 66s, status success
L4 Process cloudlens, execution time 66s, status success
1 Process firefox, execution time 8762s, status success
2 Process firefox, execution time 8762s, status success
L1 Process firefox, execution time 8762s, status success
L2 Process firefox, execution time 8762s, status success
3 Process firefox, execution time 8762s, status success
L3 Process firefox, execution time 8762s, status success
L4 Process firefox, execution time 8762s, status success
1 Process eclipse, execution time 2390s, status success
2 Process eclipse, execution time 2390s, status success
L1 Process eclipse, execution time 2390s, status success
L2 Process eclipse, execution time 2390s, status success
3 Process eclipse, execution time 2390s, status success
L3 Process eclipse, execution time 2390s, status success
L4 Process eclipse, execution time 2390s, status success
1 Process cloudlens, execution time 2s, status failure
2 Process cloudlens, execution time 2s, status failure
L1 Process cloudlens, execution time 2s, status failure
L2 Process cloudlens, execution time 2s, status failure
3 Process cloudlens, execution time 2s, status failure
L3 Process cloudlens, execution time 2s, status failure
L4 Process cloudlens, execution time 2s, status failure
1 Process cloudlens, execution time 5s, status sucess
2 Process cloudlens, execution time 5s, status sucess
L1 Process cloudlens, execution time 5s, status sucess
L2 Process cloudlens, execution time 5s, status sucess
3 Process cloudlens, execution time 5s, status sucess
L3 Process cloudlens, execution time 5s, status sucess
L4 Process cloudlens, execution time 5s, status sucess
1 Bye!
2 Bye!
L1 Bye!
L2 Bye!
3 Bye!
L3 Bye!
L4 Bye!
